scaladoc-js/src/versions-dropdown/DropdownHandler.scala

```diff
@@ -0,0 +1,94 @@
+package dotty.tools.scaladoc
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Success,Failure}
+
+import org.scalajs.dom._
+import org.scalajs.dom.ext._
+import scala.scalajs.js.annotation.JSExportTopLevel
+import org.scalajs.dom.ext.Ajax
+import scala.scalajs.js
+import scala.scalajs.js.JSON
+
+trait Versions extends js.Object:
+  def versions: js.Dictionary[String]
+
+class DropdownHandler:
+
+  val KEY = "versions-json"
+  val UNDEFINED_VERSIONS = "undefined_versions"
+
+  private def addVersionsList(json: String) =
+    val ver = JSON.parse(json).asInstanceOf[Versions]
+    val ddc = document.getElementById("dropdown-content")
+    for (k, v) <- ver.versions do
+      var child = document.createElement("a").asInstanceOf[html.Anchor]
+      child.href = v
+      child.text = k
+      ddc.appendChild(child)
+    val arrow = document.createElement("span").asInstanceOf[html.Span]
+    arrow.classList.add("ar")
+    document.getElementById("dropdown-button").appendChild(arrow)
+
+  private def disableButton() =
+    val btn = document.getElementById("dropdown-button").asInstanceOf[html.Button]
+    btn.disabled = true
+    btn.classList.remove("dropdownbtnactive")
+
+  private def getURLContent(url: String): Future[String] = Ajax.get(url).map(_.responseText)
+
+  window.sessionStorage.getItem(KEY) match
+    case null => // If no key, returns null
+      try
+        getURLContent(Globals.versionsDictionaryUrl).onComplete {
+          case Success(json: String) =>
+            window.sessionStorage.setItem(KEY, json)
+            addVersionsList(json)
+          case Failure(_) =>
+            window.sessionStorage.setItem(KEY, UNDEFINED_VERSIONS)
+            disableButton()
+        }
+      catch // Globals.versionDictionaruUrl is undefined
```

Comment:
The exception is not thrown by `getUrlContent` but by accessing `versionsDictionaryUrl` itself.
When I want to access the js proprty that actually does not exists (because not always we want to have more docs linked) we do not append js property to html header properties, thus when we want to access it we get and exception at js runtime. I don't know if I can handle this one better, I did it after inspecting the transpiled js code and it's the only solution I could come up with.

Reply:
Would a `if Globals.versionDictionaryUrl == js.undefined` work?

Reply:
I will confirm that, but if I recall correctly calling `Globals.versionDictionaryUrl` throws the exception, not the comparison

Reply:
I changed the type of `versionDictionaryUrl` to `js.UndefOr[String]` then did your comparision. As I said the exception is thrown by accessing the property at js runtime so I wrapped it with try catch
![obraz](https://user-images.githubusercontent.com/32793002/121187381-4b468980-c868-11eb-85d3-f3fb25de4723.png)

Reply:
In addition to changing it to `js.UndefOr[String]`, use
```scala
if (js.typeOf(Globals.versionDictionaryUrl) != "undefined") {
  // it's defined
} else {
  // it's not
}
```

Reply:
Thanks, will try out